akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorSpec.scala

```diff
@@ -247,26 +262,30 @@ object EventSourcedBehaviorSpec {
                                                         .thenStop
 
                                                     case Fail =>
-                                                      throw new TestException("boom!")
+                                                      throw TestException("boom!")
 
                                                     case StopIt =>
                                                       Effect.none.thenStop()
 
                                                   },
-                                                eventHandler = (state, evt) ⇒
+                                                eventHandler = (state, evt) =>
                                                   evt match {
-                                                    case Incremented(delta) ⇒
+                                                    case Incremented(delta) =>
                                                       probe ! ((state, evt))
                                                       State(state.value + delta, state.history :+ state.value)
-                                                  }).receiveSignal {
-      case RecoveryCompleted(_) ⇒ ()
-      case SnapshotCompleted(_) ⇒
-        snapshotProbe ! Success(Done)
-      case SnapshotFailed(_, failure) ⇒
-        snapshotProbe ! Failure(failure)
-    }
+                                                  })
+      .receiveSignal {
+        case RecoveryCompleted(_) ⇒ ()
+        case SnapshotCompleted(_) ⇒
+          snapshotProbe ! Success(Done)
+        case SnapshotFailed(_, failure) ⇒
+          snapshotProbe ! Failure(failure)
+       case e: RetentionSignal =>
+         // Consistenly receive expected events here, but
+         // the ref in the test itself never receives it in `expect`
+         retentionProbe ! e
```

Comment:
that is because `RetentionSignal` is a `Signal` so it will not be received as a message even though you defined the message type of the probe to be RetentionSignal. You need to wrap it into some other custom message

Reply:
Created issue https://github.com/akka/akka/issues/26535
I have also done this mistake.

Reply:
I knew it was something right in front of me. Thanks!